### ISSUE TITLE
Ensure Ursula calls kfrag.verify()

### ISSUE THREAD

**Issue by cygnusv** [nucypher/nucypher#167]
For each kfrag an Ursula receives, she must call `kfrag.verify()` to check its validity. 

BTW, what happens if it fails?

**Comment by KPrasch** [comment]
@cygnusv - Can this be closed?

**Comment by cygnusv** [comment]
I don't think so. `ProxyRESTServer.set_policy()` should validate KFrags before saving them in the Data Store.

**Comment by KPrasch** [comment]
@jMyles - This seems relevant to #395
